### PR TITLE
Added an SNS forwarder to pass DynamoDB stream data to SNS.

### LINKED ISSUE #75: Move off of Kinesis and onto SQS Lambda for event triggers
Need to move off of Kinesis, and onto SQS.

### PR DESCRIPTION
**Pull request by mikegrima**
This will be the new default way that the Collector, Differ and subsequent lambdas
will be triggered (instead of directly getting triggered off of the DDB stream)

Fixes #75

### CODE REVIEW

File: historical/common/dynamodb.py [hunk 1]
```diff
@@ -99,54 +93,73 @@ def deserialize_current_dynamo_to_pynamo(record, model):
     :param model:
     :return:
     """
+    # Was the item in question too big for SNS? If so, then we need to fetch the item from the current Dynamo table:
+    if record.get("lol_too_big"):
```
Comment (kevgliss) [inline historical/common/dynamodb.py]: Really?

File: historical/common/dynamodb.py [hunk 2]
```diff
@@ -99,54 +93,73 @@ def deserialize_current_dynamo_to_pynamo(record, model):
...
 
-def process_dynamodb_differ_record(record, durable_model, diff_func=None, region_attr="Region"):
+def process_dynamodb_differ_record(record, current_model, durable_model, diff_func=None):
     """
-    Processes a group of DynamoDB NewImage records (for Differ events).
+    Processes a DynamoDB NewImage record (for Differ events).
 
-    This will ONLY process the record if the record exists in one of the regions defined in the
-    DIFF_REGIONS environment variable.
+    This will ONLY process the record if the record exists in one of the regions defined by the SNSZR_REGIONS of the
```
Comment (kevgliss) [inline historical/common/dynamodb.py]: Its not clear via the variable name what the `ZR` part or `SNSZR_REGIONS` is.

File: historical/common/sns.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+@retry(stop_max_attempt_number=4, wait_exponential_multiplier=1000, wait_exponential_max=1000)
+def _publish_message(client, blob, topic_arn):
+    client.publish(TopicArn=topic_arn, Message=blob)
+
+
+def shrink_sns_blob(record):
+    """
+    Makes a shrunken blob to be sent to SNS (due to size limitations of SNS messages).
+    This will essentially remove the "configuration" field such that
```
Comment (kevgliss) [inline historical/common/sns.py]: Such that what?

File: historical/common/sns.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+def process_sns_forward(record, topic_arn, client):
+    """
+    Prepares and ships an individual record over to SNS for future processing.
+    :param record:
+    :param topic_arn:
+    :param client:
+    :return:
+    """
+    print(record)
```
Comment (kevgliss) [inline historical/common/sns.py]: Un-needed print.

File: historical/common/dynamodb.py (hunk 2 above)
Comment (mikegrima) [reply historical/common/dynamodb.py]: Renaming it to `SQSProxy`.

File: historical/common/sns.py (hunk 3 above)
Comment (mikegrima) [reply historical/common/sns.py]: Fixing.  It's such that the SNS message size is under 256 KB.

File: historical/common/sns.py (hunk 4 above)
Comment (mikegrima) [reply historical/common/sns.py]: Fixing.

File: historical/common/dynamodb.py (hunk 1 above)
Comment (mikegrima) [reply historical/common/dynamodb.py]: It's the LTB flag.

**Pull request merged**